tests/test_search.py

```diff
@@ -418,17 +404,19 @@ def check_count(self, es_query, expected_count, timeout=5):
             self.fail("elasticsearch failed to return all results.")
 
 
-class TestGCPSearch(TestSearchBase, unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        super().search_setup(dss.Replica.gcp)
+class TestGCPSearch(TestSearchBase):
+    replica = dss.Replica.gcp
 
 
-class TestAWSSearch(TestSearchBase, unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        super().search_setup(dss.Replica.aws)
+class TestAWSSearch(TestSearchBase):
+    replica = dss.Replica.aws
+
 
+# Prevent unittest's discovery from attempting to discover the base test class. The alterative, not inheriting
+# TestCase in the base class, is too inconvenient because it interferes with auto-complete and generates PEP-8
+# warnings about the camel case methods.
+#
+del TestSearchBase
```

Comment:
an alternative would be to move TestSearchBase into another file called searchBase.py and import it into test_search.

Reply:
That would be an alternative, but I am not convinced a desirable one. In either case, I'd rather not mix a code moving change into this PR for obvious reasons. With this particular change, these tests finally become usable in PyCharm. I can right click a failed test an rerun it, something that wasn't possible before.